Mathlib/Combinatorics/SimpleGraph/ConcreteColorings.lean

```diff
@@ -48,4 +48,28 @@ theorem chromaticNumber_pathGraph (n : ℕ) (h : 2 ≤ n) :
   · simpa only [pathGraph_two_eq_top, chromaticNumber_top] using
       chromaticNumber_mono_of_embedding (pathGraph_two_embedding n h)
 
+theorem coloring_apply_iff_of_bicoloring_of_walk_general {α} (G : SimpleGraph α)
```

Comment:
Maybe this?
```suggestion
theorem Coloring.bicoloring_apply_iff_of_walk {α} {G : SimpleGraph α}
```
The `_general` is against the mathlib naming convention for theorems. Using `bicoloring` here is pushing it a little, though if we add this word to the module docs in this module it's fine.

Putting it into the `Coloring` namespace means you can use dot notation on `c` to access it. I also made the `G` argument implicit since it's implied by both `c` and `p`.